# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmValidate.cpp
# Commit: aca2e6cff6b9
# Full Hash: aca2e6cff6b9c66232538f82538d3d1965be4eca
# Author: Andy Wingo <wingo@igalia.com>
# Date: 2019-10-22 21:43:14
# Regressor Bug: 1578418
# File Overlap Count: 1
# Description:
#   Bug 1578418 - Use WasmABIResults iterator to place block and function results r=luke,lth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44477
# ==============================================================================

diff -r e21ad27bfd0a -r aca2e6cff6b9 js/src/wasm/WasmValidate.cpp
--- a/js/src/wasm/WasmValidate.cpp	Tue Oct 22 10:57:43 2019 +0000
+++ b/js/src/wasm/WasmValidate.cpp	Tue Oct 22 09:04:59 2019 +0000
@@ -578,13 +578,13 @@
                               &nothing));
       }
       case uint16_t(Op::Block):
-        CHECK(iter.readBlock());
+        CHECK(iter.readBlock(&unusedType));
       case uint16_t(Op::Loop):
-        CHECK(iter.readLoop());
+        CHECK(iter.readLoop(&unusedType));
       case uint16_t(Op::If):
-        CHECK(iter.readIf(&nothing));
+        CHECK(iter.readIf(&unusedType, &nothing));
       case uint16_t(Op::Else):
-        CHECK(iter.readElse(&unusedType, &nothings));
+        CHECK(iter.readElse(&unusedType, &unusedType, &nothings));
       case uint16_t(Op::I32Clz):
       case uint16_t(Op::I32Ctz):
       case uint16_t(Op::I32Popcnt):
